cdap-app-fabric/src/main/java/org/apache/hadoop/fs/ftp/FTPInputStream.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.hadoop.fs.ftp;
+
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.hadoop.classification.InterfaceAudience;
+import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.hadoop.fs.FSInputStream;
+import org.apache.hadoop.fs.FileSystem;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * {@link FTPInputStream}, copied from Hadoop and modified, that doesn't throw an exception when seeks are attempted
+ * to the current position. Position equality check logic in {@link FTPInputStream#seek}  is the only change from the
+ * original class in Hadoop.
+ */
+@InterfaceAudience.Private
+@InterfaceStability.Unstable
+public class FTPInputStream extends FSInputStream {
+
+  InputStream wrappedStream;
+  FTPClient client;
+  FileSystem.Statistics stats;
+  boolean closed;
+  long pos;
+
+  public FTPInputStream(InputStream stream, FTPClient client,
+                        FileSystem.Statistics stats) {
+    if (stream == null) {
+      throw new IllegalArgumentException("Null InputStream");
+    }
+    if (client == null || !client.isConnected()) {
+      throw new IllegalArgumentException("FTP client null or not connected");
+    }
+    this.wrappedStream = stream;
+    this.client = client;
+    this.stats = stats;
+    this.pos = 0;
+    this.closed = false;
+  }
+
+  @Override
+  public long getPos() throws IOException {
+    return pos;
+  }
+
+  // We don't support seek.
```

Comment:
just update the comment here, as we now don't throw exception if ``getPos() == pos``